device-server/src/main/kotlin/com/badoo/automation/deviceserver/host/SimulatorsNode.kt

```diff
@@ -400,8 +400,17 @@ class SimulatorsNode(
         return getDeviceFor(deviceRef).dataContainer(dataPath.bundleId).readFile(dataPath.path)
     }
 
-    override fun pushFile(ref: DeviceRef, fileName: String, data: ByteArray, bundleId: String) {
-        getDeviceFor(ref).dataContainer(bundleId).writeFile(File(fileName), data)
+    override fun pushFile(ref: DeviceRef, fileName: String, data: ByteArray, bundleId: String, path: String?) {
+        getDeviceFor(ref).dataContainer(bundleId).writeFile(File(fileName), data, path)
+    }
+
+    override fun deleteFile(ref: DeviceRef, fileName: String, bundleId: String, path: String?) {
+        val file = if (path.isNullOrBlank()) {
+            File(getDeviceFor(ref).dataContainer(bundleId).basePath.absolutePath, fileName)
```

Comment:
Maybe let's create another abstraction (like SharedResources or something). 
because getting `.basePath` from `dataContainer` looks a bit like a hack